croptalk/model_llm.py

```diff
@@ -147,10 +148,23 @@ def create_chain(
     retriever_chain = create_retriever_chain(basic_llm, document_retriever)
     tool_chain = create_tool_chain(basic_llm)
 
+    chain = (
+            PromptTemplate.from_template(ROUTE_TEMPLATE) | answer_llm | StrOutputParser()
+    )
+
+    def route(info):
+        if "document_retrieval" in info["topic"].lower():
+            return tool_chain
+        else:
+            return retriever_chain
```

Comment:
Let's discuss when you can... Maybe it's just not clear in my head, but it feels a bit weird to me here: if **document_retrieval** then we call tool_chain, and otherwise we call **retriever_chain**...